### PR TITLE
[Feature][Connector-V2][Oss jindo] Add oss jindo source & sink connector

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

sub-task of #3018 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
spark integration test:

text write:

image

orc write:

image

parquet write:

image

json write:

image

flink integration test:

text write:

image

orc write:

image

parquet write:

image

json write:

image

**Review by hailin0 — COMMENTED** [review]
Please update check user docs

File: seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/oss/sink/OssFileSinkFactory.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@AutoService(Factory.class)
+public class OssFileSinkFactory implements TableSinkFactory {
+    @Override
+    public String factoryIdentifier() {
+        return FileSystemType.OSS.getFileSystemPluginName();
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/oss/sink/OssFileSinkFactory.java:38]: Please add UT about option rule.

File: seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/oss/source/OssFileSourceFactory.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@AutoService(Factory.class)
+public class OssFileSourceFactory implements TableSourceFactory {
+    @Override
+    public String factoryIdentifier() {
+        return FileSystemType.OSS.getFileSystemPluginName();
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/oss/source/OssFileSourceFactory.java:39]: Please add UT for this option rule.

**Review by EricJoy2048 — APPROVED** [review]

**Comment by EricJoy2048** [comment]
I found a bug from this CI. https://github.com/apache/incubator-seatunnel/issues/3611

**Review by TaoZex — APPROVED** [review]
+1

File: seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/pom.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        <jindo-sdk.version>4.6.1</jindo-sdk.version>
+        <hadoop-common.version>2.9.2</hadoop-common.version>
+    </properties>
+
+    <repositories>
+        <repository>
+            <id>jindodata</id>
+            <url>https://jindodata-binary.oss-cn-shanghai.aliyuncs.com/mvn-repo/</url>
+        </repository>
+    </repositories>
```
Comment (wuchunfu) [inline seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/pom.xml:42]: I'm not sure if this kind of warehouse is compliant, @CalvinKirs please help me to have a look.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/pom.xml:42]: This sdk only posted in this repository😂, @CalvinKirs PTAL

Comment (CalvinKirs) [reply seatunnel-connectors-v2/connector-file/connector-file-oss-jindo/pom.xml:42]: I think ok (if everone can visit this link)

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**